### ISSUE TITLE
Review/fix payments headers and other HTML semantics

### ISSUE THREAD

**Issue by LZoog** [mozilla/fxa#2962]
This is how our Subscriptions page outline currently builds as:

![image](https://user-images.githubusercontent.com/13018240/66942261-4a674100-f00e-11e9-8d27-c07769848037.png)

We should reevaluate how we want to organize the page. This can be up for discussion, but it would make sense to me if:
- Name and e-mail are _not_ headers, as they don't have any additional text following and don't seem like top-level indicators to me, likely also not rendering these in a `<header>` tag
- Firefox Fortress / Guardian (product names) should be H3's under "Subscriptions" (an H2), and "Billing Information" should become an H4

In addition, I noticed the breadcrumbs `ol`, while functioning as navigation, is not in a `nav` tag.

The above are not all encompassing, part of this issue is exploring if anything else within payments should be updating tag-wise.

**Comment by lmorchard** [comment]
Yeah, the current state of affairs is probably me just throwing markup at the wall and calling it good. Any improvements here would be welcome
